### DISCUSSION TITLE
Which Hyperparameters to tune?

### DISCUSSION THREAD

**Discussion by ourownstory** [ourownstory/neural_prophet#527] [Q&A - get help using NeuralProphet] [answered]
Which hyperparameters should I leave on auto and which should I tune, if any?

**Comment by ourownstory** [comment, marked as answer]
It depends. What are you forecasting and what are the data's characteristics?

Without knowing about that, here are some quick pointers:
- Auto-Regression:
-- Y/N and N lags
-- Hidden layers: Y/N and N layers
- (Trend: N changepoints)
- (Seasonality: daily/weekly/yearly/other)
- (Holidays: which county)
- Any future regressors?
- any lagged regressors? (if so, hidden layers?)

Hyperparameters should be tuned if hidden layers (NN used):
- learning rate
- (epochs, batch size)

Further, if you have a panel dataset - you can train one global model with shared weights across all or train one model per time-series.
